api/lib/client.js

```diff
@@ -142,18 +141,16 @@ Client.prototype._error = function (error, resData, callback) {
   }
 }
 
-Client.prototype._redirect = function (response) {
-  var parts
-
+Client.prototype._redirect = function (response, callback) {
   if (!/^https?:/.test(response.headers.location)) {
-    response.headers.location = urlmodule.resolve(options.url, response.headers.location)
+    response.headers.location = urlmodule.resolve(this.options.url, response.headers.location)
   }
 
   // Parse location to check for port
-  parts = urlmodule.parse(response.headers.location)
-  if (parts.protocol == 'http:') {
-    options.secure = false
-    options.port = parts.port || 80
+  var parts = urlmodule.parse(response.headers.location)
+  if (parts.protocol === 'http:') {
+    this.options.secure = false
+    this.options.port = parts.port || 80
```

Comment:
:+1: for completely deprecating and removing this part of the api- in a different PR ;)